src/test/java/com/myhome/services/unit/UserSDJpaServiceTest.java

```diff
@@ -0,0 +1,189 @@
+package com.myhome.services.unit;
+
+import com.myhome.controllers.dto.UserDto;
+import com.myhome.controllers.dto.mapper.UserMapper;
+import com.myhome.domain.Community;
+import com.myhome.domain.CommunityAdmin;
+import com.myhome.domain.User;
+import com.myhome.repositories.UserRepository;
+import com.myhome.services.CommunityService;
+import com.myhome.services.springdatajpa.UserSDJpaService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+class UserSDJpaServiceTest {
+
+  private String USER_ID = "test-user-id";
+  private String USER_NAME = "test-user-id";
+  private String USER_EMAIL = "test-user-id";
+  private String USER_PASSWORD = "test-user-id";
+
+  @Mock
+  private UserRepository userRepository;
+  @Mock
+  private UserMapper userMapper;
+  @Mock
+  private PasswordEncoder passwordEncoder;
+  @Mock
+  private CommunityService communityService;
+  @InjectMocks
+  private UserSDJpaService userService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  void createUserSuccess() {
+    // given
+    UserDto request = getDefaultUserDtoRequest();
+    User resultUser = new User(request.getName(), request.getUserId(), request.getEmail(), request.getEncryptedPassword());
+    UserDto response = new UserDto(resultUser.getId(), resultUser.getUserId(), resultUser.getName(), resultUser.getEmail(), null, resultUser.getEncryptedPassword(), new HashSet<>());
+
+    given(userRepository.findByEmail(request.getEmail()))
+        .willReturn(null);
+    given(passwordEncoder.encode(request.getPassword()))
+        .willReturn(request.getPassword());
+    given(userMapper.userDtoToUser(request))
+        .willReturn(resultUser);
+    given(userRepository.save(resultUser))
+        .willReturn(resultUser);
+    given(userMapper.userToUserDto(resultUser))
+        .willReturn(response);
+
+    // when
+    Optional<UserDto> createdUserDto = userService.createUser(request);
+
+    // then
+    assertTrue(createdUserDto.isPresent());
+    assertEquals(createdUserDto.get(), response);
```

Comment:
`createdUserDto.get()` can be extracted to the variable as it is used twice in this test context.

Please check other test methods - same change is needed there